### PR TITLE
Pingdom connector

### PR DESCRIPTION
**Pull request by rameshlk**

### CODE REVIEW

File: basic_auth/pingdom_connector.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      ]
+      },
+
+    execute: ->(connection, input) {
+        get("https://api.pingdom.com/api/2.0/checks/#{input['checkid']}")
+      },
+
+    output_fields: ->() {
+        [ 
+          {name: 'check',type: :object,properties:[
```
Comment (eeshan92) [inline basic_auth/pingdom_connector.rb]: Let's move this into object_definitions

File: basic_auth/pingdom_connector.rb [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        ]
+      },
+
+        poll: ->(connection, input, page) {
+          limit = 100
+          page ||= 0
+        	created_since = (input['since'] || Time.now).to_i
+					offset = (limit * page)
+        	response = get("https://api.pingdom.com/api/2.0/actions?from=#{created_since}&limit=100&offset=#{offset}")                
+        	next_created_since = response['actions']['alerts'].last['time'] if response['actions']['alerts'].present?
```
Comment (eeshan92) [inline basic_auth/pingdom_connector.rb]: `next_created_since` is not used

File: basic_auth/pingdom_connector.rb [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+          page = page + 1
```
Comment (eeshan92) [inline basic_auth/pingdom_connector.rb]: We don't always want to poll the next page. Should increment this only if there is a next page to poll. Example: `response['actions']['alerts'].length == limit`

File: basic_auth/pingdom_connector.rb [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+          events: response['actions']['alerts'],
+          next_page: page
+        }
+      },
+       sort_by: ->(response) {
+         response['time']
+      },
+
+        output_fields: ->() {
+         [
```
Comment (eeshan92) [inline basic_auth/pingdom_connector.rb]: can be moved into object_definitions

File: basic_auth/pingdom_connector.rb [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+      },
+    
+      {
+        name: 'username',
+        optional: true,
+        hint: 'Your username'
+      },
+      {
+        name: 'password',
+        control_type: 'password',
```
Comment (kandadaboggu) [inline basic_auth/pingdom_connector.rb]: Don't we need?
```
optional: false
```

**Comment by kandadaboggu** [comment]
Fix all the formatting issues

**Comment by rameshlk** [comment]
I am closing this PR,this is already merged in workato project.

**Pull request closed without merging**